### PR TITLE
feat: add custom Dockerfile for Ubuntu 20 and Ubuntu 18 improvements

### PR DESCRIPTION
**Pull request by ilya-naumov**

### CODE REVIEW

File: dockerfiles/README.md [hunk 1]
```diff
@@ -90,7 +90,9 @@ docker build --build-arg package_url=https://storage.openvinotoolkit.org/reposit
 ----------------
 
 * Custom image with CPU, iGPU, VPU support  
-To learn more, follow [Build custom Intel® Distribution of OpenVINO™ toolkit Docker image](ubuntu18/build_custom/README.md).
+We prepared Dockerfiles for custom development OpenVINO™ from source. To learn more, follow:
```
Comment (avladimi) [inline dockerfiles/README.md:93]:
```suggestion
Use Dockerfiles to develop a custom version of OpenVINO™ from source. To learn more, follow:
```

File: dockerfiles/ubuntu18/build_custom/README.md [hunk 2]
```diff
@@ -22,6 +22,7 @@ docker build --no-cache -t image:tag .
 ```
 
 You can use docker `--build-arg` option to override the following variables:  
+`OPENVINO_FORK` - To specify GitHub fork of OpenVINO repository to use. By default, it is main OpenVINO repository.
```
Comment (avladimi) [inline dockerfiles/ubuntu18/build_custom/README.md:25]:
```suggestion
`OPENVINO_FORK` - To specify a GitHub fork of the OpenVINO repository to use. By default, it is main OpenVINO repository.  
```

Comment (avladimi) [reply dockerfiles/ubuntu18/build_custom/README.md:25]: Please also fix line 26: To specify branches with source code.

File: dockerfiles/ubuntu20/build_custom/README.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
+# Build custom Intel® Distribution of OpenVINO™ toolkit Docker image
+This repository folder contains Dockerfile to build a docker image with the Intel® Distribution of OpenVINO™ toolkit.
```
Comment (avladimi) [inline dockerfiles/ubuntu20/build_custom/README.md:2]:
```suggestion
This repository folder contains the Dockerfile to build a docker image with the Intel® Distribution of OpenVINO™ toolkit.  
```

File: dockerfiles/ubuntu20/build_custom/README.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+Go to the folder with the Dockerfile and run:
+```
+docker build -t [image:tag] .
+```
+
+* `/opt/intel/openvino` folder will contain OpenVINO build
+* `/opt/intel/repo` will contain OpenVINO™ and OpenCV git repositories in `openvino` and `opencv` folders accordingly.
+
+
+If you want to rebuild the entire image, use docker `--no-cache` option:
```
Comment (avladimi) [inline dockerfiles/ubuntu20/build_custom/README.md:19]:
```suggestion
If you want to rebuild the entire image, use the docker `--no-cache` option:
```

File: dockerfiles/ubuntu20/build_custom/README.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+```
+docker build --no-cache -t image:tag .
+```
+
+You can use docker `--build-arg` option to override the following variables:
```
Comment (avladimi) [inline dockerfiles/ubuntu20/build_custom/README.md:24]:
```suggestion
You can use the docker `--build-arg` option to override the following variables:  
```

File: dockerfiles/ubuntu20/build_custom/README.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+* `/opt/intel/repo` will contain OpenVINO™ and OpenCV git repositories in `openvino` and `opencv` folders accordingly.
+
+
+If you want to rebuild the entire image, use docker `--no-cache` option:
+```
+docker build --no-cache -t image:tag .
+```
+
+You can use docker `--build-arg` option to override the following variables:  
+`OPENVINO_FORK` - To specify GitHub fork of OpenVINO repository to use. By default, it is main OpenVINO repository.
```
Comment (avladimi) [inline dockerfiles/ubuntu20/build_custom/README.md:25]:
```suggestion
`OPENVINO_FORK` - To specify a GitHub fork of the OpenVINO repository to use. By default, it is main OpenVINO repository.  
```

File: dockerfiles/ubuntu20/build_custom/README.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+
+If you want to rebuild the entire image, use docker `--no-cache` option:
+```
+docker build --no-cache -t image:tag .
+```
+
+You can use docker `--build-arg` option to override the following variables:  
+`OPENVINO_FORK` - To specify GitHub fork of OpenVINO repository to use. By default, it is main OpenVINO repository.  
+`OPENVINO_BRANCH`, `OPENCV_BRANCH`, `OMZ_BRANCH` - To specify a branches with source code. By default, they are equal to "master".
```
Comment (avladimi) [inline dockerfiles/ubuntu20/build_custom/README.md:26]:
```suggestion
`OPENVINO_BRANCH`, `OPENCV_BRANCH`, `OMZ_BRANCH` - To specify branches with source code. By default, they are equal to "master".  
```

File: dockerfiles/ubuntu20/build_custom/README.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+`OPENVINO_BRANCH`, `OPENCV_BRANCH`, `OMZ_BRANCH` - To specify a branches with source code. By default, they are equal to "master".  
+`OCL_VERSION` - To specify the version of Intel® Graphics Compute Runtime for OpenCL™ Driver on Linux. By default, it is equal to "19.41.14441".
+
+**For example**:  
+This command builds an image with OpenVINO™ 2021.2 release.
+```
+docker build -t openvino:2021.2 --build-arg OPENVINO_BRANCH="releases/2021/2" .
+```
+
+You can manually setup cmake parameters to build a custom package from source code using these files:
```
Comment (avladimi) [inline dockerfiles/ubuntu20/build_custom/README.md:35]:
```suggestion
You can manually set up cmake parameters to build a custom package from source code using these files:  
```

File: dockerfiles/ubuntu20/build_custom/README.md [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+By default, these files already contain some parameters for *Debug* build  
+Do not override PATH/PREFIX options. This can break a build of package.
+
+### Build stages
+The docker image is built using a multi-step build:
+1. **setup_openvino**  
+    Clone OpenVINO™ git repository with submodules and install build dependencies.
+2. **build_openvino**  
+    Build OpenVINO™ (CPU, iGPU, VPU support) with the parameters specified in openvino_cmake.txt.  
+    It doesn't include OpenCV and Open Model Zoo.
```
Comment (avladimi) [inline dockerfiles/ubuntu20/build_custom/README.md:49]:
```suggestion
    It does not include OpenCV and Open Model Zoo.
```

File: dockerfiles/ubuntu20/build_custom/README.md [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+3. **copy_openvino**  
+    Copy OpenVINO™ build to clear Ubuntu:18.04 image.
+4. **openvino**  
+    Install OpenVINO™ dependencies. Now you can use it.
+5. **opencv**  
+    Build and setup OpenCV with the parameters specified in opencv_cmake.txt.
+6. **open_model_zoo**  
+    Clone and setup Open Model Zoo repository.
```
Comment (avladimi) [inline dockerfiles/ubuntu20/build_custom/README.md:57]:
```suggestion
    Clone and set up Open Model Zoo repository.
```

File: dockerfiles/ubuntu20/build_custom/README.md [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+
+Use docker `--target` option to specify a final stage.
```
Comment (avladimi) [inline dockerfiles/ubuntu20/build_custom/README.md:59]:
```suggestion
Use the docker `--target` option to specify a final stage.
```

File: dockerfiles/ubuntu20/build_custom/README.md [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+```
+python3 docker_openvino.py test -t [image:tag] -dist custom 
+```
+
+>**Note**:  
+Docker CI framework automatically runs the corresponding tests.  
+By default, the product version is equal to the latest release version. Use `-p` to override this.
+
+## How to run
+Please follow [Run built image](../get-started.md#run-built-image) section in Docker CI getting started guide.
```
Comment (avladimi) [inline dockerfiles/ubuntu20/build_custom/README.md:81]:
```suggestion
Please follow the [Run built image](../get-started.md#run-built-image) section in Docker CI getting started guide.
```

File: docs/tutorials/OV_docker_VScode.md [hunk 13]
```diff
@@ -4,7 +4,7 @@
 
 The purpose of this tutorial is to examine [Intel® Distribution of OpenVINO™ toolkit](https://software.intel.com/openvino-toolkit) development containers.
 
-This tutorial will go step-by-step to demonstrate how to set up an OpenVINO™ [custom development container](https://github.com/openvinotoolkit/docker_ci/tree/master/dockerfiles/ubuntu18/build_custom) under Visual Studio Code* to use it as your full-time development environment.
+This tutorial will go step-by-step to demonstrate how to set up an OpenVINO™ custom development container under Visual Studio Code* to use it as your full-time development environment.
```
Comment (avladimi) [inline docs/tutorials/OV_docker_VScode.md:7]:
```suggestion
This tutorial provides step-by-step instructions to demonstrate how to set up an OpenVINO™ custom development container under Visual Studio Code* to use it as your full-time development environment.
```

File: docs/tutorials/OV_docker_VScode.md [hunk 14]
```diff
@@ -19,7 +19,7 @@ This tutorial requires the following:
 
 Intel® Distribution of OpenVINO™ toolkit is produced from different open source and closed source repositories and components. As well OpenVINO has a couple of dependencies which need to be present on your computer. Additionally, to install some of them, you need to have root/admin rights. This might not be desirable. Using Docker and Visual Studio Code integration represents a much cleaner way of development. 
 
-We prepared a [Dockerfile for custom development](https://github.com/openvinotoolkit/docker_ci/tree/master/dockerfiles/ubuntu18/build_custom) OpenVINO™ from source. It includes the following components:
+We prepared Dockerfiles for custom development OpenVINO™ from source based on [Ubuntu 18](https://github.com/openvinotoolkit/docker_ci/tree/master/dockerfiles/ubuntu18/build_custom) and [Ubuntu 20](https://github.com/openvinotoolkit/docker_ci/tree/master/dockerfiles/ubuntu20/build_custom). They include the following components:
```
Comment (avladimi) [inline docs/tutorials/OV_docker_VScode.md:22]:
```suggestion
We prepared Dockerfiles for custom development of OpenVINO™ from source based on [Ubuntu 18](https://github.com/openvinotoolkit/docker_ci/tree/master/dockerfiles/ubuntu18/build_custom) and [Ubuntu 20](https://github.com/openvinotoolkit/docker_ci/tree/master/dockerfiles/ubuntu20/build_custom). They include the following components:
```

File: docs/tutorials/OV_docker_VScode.md [hunk 15]
```diff
@@ -31,7 +31,7 @@ You can customize Cmake* build parameters for OpenVINO itself and OpenCV as well
 
 ## Build OpenVINO custom development Docker
 The first step, we need to build a OpenVINO™ Docker image, by default it uses Debug mode and master branch for all repositories.
-Lets go to Docker CI repository local copy and run `docker build` from `docker_ci/dockerfiles/ubuntu18/build_custom` folder:
+Lets go to Docker CI repository local copy and run `docker build` from `docker_ci/dockerfiles/<os>/build_custom` folder:
```
Comment (avladimi) [inline docs/tutorials/OV_docker_VScode.md:34]:
```suggestion
Go to the Docker CI repository local copy and run `docker build` from `docker_ci/dockerfiles/<os>/build_custom` folder:
```

File: dockerfiles/README.md (hunk 1 above)
Comment (ilya-naumov) [reply dockerfiles/README.md:93]: May be replace it to something like this:

You can use Dockerfiles from `build_custom` folders to build a custom version of OpenVINO™ from source codes for development

**Pull request merged**